skfda/exploratory/visualization/representation.py

```diff
@@ -79,249 +132,402 @@ def _get_color_info(fdata, group, group_names, group_colors, legend, kwargs):
-            same number of points will be used in the two axes. By default
-            in unidimensional plots will be used 501 points; in surfaces
-            will be used 30 points per axis, wich makes a grid with 900
-            points.
-        domain_range (tuple or list of tuples, optional): Range where the
-            function will be plotted. In objects with unidimensional domain
-            the domain range should be a tuple with the bounds of the
-            interval; in the case of surfaces a list with 2 tuples with
-            the ranges for each dimension. Default uses the domain range
-            of the functional object.
-        group (list of int): contains integers from [0 to number of
-            labels) indicating to which group each sample belongs to. Then,
-            the samples with the same label are plotted in the same color.
-            If None, the default value, each sample is plotted in the color
-            assigned by matplotlib.pyplot.rcParams['axes.prop_cycle'].
-        group_colors (list of colors): colors in which groups are
-            represented, there must be one for each group. If None, each
-            group is shown with distict colors in the "Greys" colormap.
-        group_names (list of str): name of each of the groups which appear
-            in a legend, there must be one for each one. Defaults to None
-            and the legend is not shown. Implies `legend=True`.
-        legend (bool): if `True`, show a legend with the groups. If
-            `group_names` is passed, it will be used for finding the names
-            to display in the legend. Otherwise, the values passed to
-            `group` will be used.
-        **kwargs: if dim_domain is 1, keyword arguments to be passed to
-            the matplotlib.pyplot.plot function; if dim_domain is 2,
-            keyword arguments to be passed to the
-            matplotlib.pyplot.plot_surface function.
-
-    Returns:
-        fig (figure object): figure object in which the graphs are plotted.
+        fdata: functional data set that we want to plot.
+        gradient_color_list: list of real values used to determine the color
+            in which each of the instances will be plotted.
+        max_grad: maximum value that the gradient_list can take, it will be
+            used to normalize the ``gradient_color_list``. If not
+            declared it will be initialized to the maximum value of
+            gradient_list
+        min_grad: minimum value that the gradient_list can take, it will be
+            used to normalize the ``gradient_color_list``. If not
+            declared it will be initialized to the minimum value of
+            gradient_list.
 
     """
 
-    fig, axes = _get_figure_and_axes(chart, fig, axes)
-    fig, axes = _set_figure_layout_for_fdata(fdata, fig, axes, n_rows, n_cols)
+    def __init__(
+        self,
+        fdata: FData,
+        gradient_color_list: Optional[Sequence[float]] = None,
```

Comment:
I think we need a better name. Maybe `gradient_values` or even better if we can use a name unrelated with colors (as was done with `group`). Try looking for alternatives to discuss.